jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -345,8 +345,9 @@ def user_api(self):
 
     @observe('trace_dataset_selected')
     def _trace_dataset_selected(self, msg=None):
-        if not hasattr(self, 'trace_dataset'):
+        if not hasattr(self, 'trace_dataset') or (hasattr(msg, 'new') and msg.new == ''):
```

Comment:
what message is triggering this scenario (I wonder if we should check the message type or if this could ever result in an immediate return when that isn't intended)?

```suggestion
        if not hasattr(self, 'trace_dataset') or getattr(msg, 'new', None)  == '':
```